posthog/hogql_queries/experiments/experiment_trends_statistics.py

```diff
@@ -0,0 +1,204 @@
+from rest_framework.exceptions import ValidationError
+from sentry_sdk import capture_exception
+from posthog.hogql_queries.experiments import FF_DISTRIBUTION_THRESHOLD, MIN_PROBABILITY_FOR_SIGNIFICANCE
+from posthog.hogql_queries.experiments.funnels_statistics import Probability
+from posthog.schema import ExperimentSignificanceCode, ExperimentVariantTrendsBaseStats
+from scipy.stats import gamma
+import numpy as np
+
+# Prior parameters (minimal prior knowledge)
+PRIOR_ALPHA = 1
+PRIOR_BETA = 1
+
+SAMPLE_SIZE = 10000
+
+
+def calculate_probabilities(
+    control_variant: ExperimentVariantTrendsBaseStats, test_variants: list[ExperimentVariantTrendsBaseStats]
+) -> list[float]:
+    """
+    Calculate the win probabilities for each variant in an experiment using Bayesian analysis.
+
+    This function computes the probability that each variant is the best (i.e., has the highest
+    conversion rate) compared to all other variants, including the control. It uses samples
+    drawn from the posterior distributions of each variant's conversion rate.
+
+    Parameters:
+    -----------
+    control_variant : ExperimentVariantTrendsBaseStats
+        Statistics for the control group, including count (successes) and exposure (total trials)
+    test_variants : list[ExperimentVariantTrendsBaseStats]
+        List of statistics for test variants to compare against the control
+
+    Returns:
+    --------
+    list[float]
+        A list of probabilities where:
+        - The first element is the probability that the control variant is the best
+        - Subsequent elements are the probabilities that each test variant is the best
+
+    Notes:
+    ------
+    - Uses a Bayesian approach with a Beta distribution as the posterior
+    - Assumes a minimally informative prior (alpha=1, beta=1)
+    - Draws samples from the posterior to estimate win probabilities
+
+    Example:
+    --------
+    >>> control = ExperimentVariantTrendsBaseStats(key="control", count=100, exposure=1000)
+    >>> test = ExperimentVariantTrendsBaseStats(key="test", count=120, exposure=1000)
+    >>> probabilities = calculate_probabilities(control, [test])
+    >>> # Returns: [0.3, 0.7] indicating the test variant is more likely to be the best
+    """
+    if len(test_variants) >= 10:
+        raise ValidationError("Can't calculate experiment results for more than 10 variants", code="too_much_data")
+    if len(test_variants) < 1:
+        raise ValidationError("Can't calculate experiment results for less than 2 variants", code="no_data")
```

Comment:
I assume we have this check elsewhere also, but maybe good to have a check for count, exposure & absolute_exposure being greater than some value? At least 0 I guess.

Reply:
This is copied over from the existing method. I'm not sure about changing the behavior here.